x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/hooks/use_action_status.tsx

```diff
@@ -37,17 +43,21 @@ export function useActionStatus(onAbortSuccess: () => void, refreshAgentActivity
         }),
       });
     }
-  }, [notifications.toasts]);
+  }, [notifications.toasts, nActions]);
 
   if (isFirstLoading) {
-    refreshActions();
+    loadActions();
   }
 
   useEffect(() => {
     if (refreshAgentActivity) {
-      refreshActions();
+      loadActions();
     }
-  }, [refreshActions, refreshAgentActivity]);
+  }, [loadActions, refreshAgentActivity]);
+
+  useEffect(() => {
+    loadActions();
+  }, [loadActions, nActions]);
```

Comment:
what is the purpose of this additional `useEffect`?

Reply:
To fetch the actions when we want more of them (after clicking "Show more"). I think it wasn't working without it, but I can check that, as I've done a lot of back and forth with the UI code.

Reply:
I would think it should work without, because `nActions` is a dependency of `loadActions`

Reply:
OK, thanks, I'll check it. Definitely don't want more `useEffect` than we need 🙂

Reply:
Should we add a comment on why the second `useEffect` is needed? I tried without it and didn't work, I don't have a better idea now than to leave it in.